client/client.go

```diff
@@ -920,6 +928,11 @@ type MetricsTimer = metrics.Timer
 // MetricsNopHandler is a noop handler that does nothing with the metrics.
 var MetricsNopHandler = metrics.NopHandler
 
+// PrepareUpdateWorkflowOperation returns an UpdateWorkflowOperation that can be used to perform Update-with-Start,
+// or an error in case the update operation is invalid. After executing Client.ExecuteWorkflow with the
+// UpdateWorkflowOperation in the start options, the update result can be obtained.
+var PrepareUpdateWorkflowOperation = internal.PrepareUpdateWorkflowOperation
```

Comment:
I had this feedback from Drew and it made sense to me:

> Also I slightly prefer Prepare as was originally sketched, as it implies that another step must follow it and gets developers to look twice at the comment above the method.  More clearly differentiates itself from a regular operation whereas New didn't stand out to me.

Reply:
I think in Go, "New" makes more sense, even if it may not in other languages, but I do not have a strong opinion, I too am fine with either

Reply:
I favor new since prepare ,to me, implies something on the server is being prepared and may lead users to believe they need to "unprepared" if they don't use an operation